### PR TITLE
feat: add search box spec 

### PR DESCRIPTION
**Pull request by eljefe223**
# Pull Request

## 📖 Description
Adds search box spec

### 🎫 Issues
Addresses: #4967 

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: specs/search/search.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+| custom icon | ![](./images/search-box-custom-icon.png)
+
+*Slot Names*
+- default - the label content
+- before-content - often times a glyph, icon, or button precedes input
+- after-content - often times a glyph, icon, or button follows the input
+- clear-content - often times a glyph, icon, or button follows that clears the input field
+
+*Host Classes*
+- disabled
```
Comment (nicholasrice) [inline specs/search/search.md:92]: Can `disabled` and `required` just be attributes and not classes? It seems folks would still have styling hooks and we would avoid class-clobbering side-effects

Comment (eljefe223) [reply specs/search/search.md:92]: Right, thanks for pointing that out, This was modeled off of `TextField` Spec, I'll remove. Probably should also remove it there.

Comment (nicholasrice) [reply specs/search/search.md]: yea that spec might be stale.

**Comment by EisenbergEffect** [comment]
Can this be closed?

**Comment by EisenbergEffect** [comment]
Seems that we already merged the component. Does this spec match what was implemented and can we merge it?

File: specs/search/search.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+- `name` - the name of the control
+- `pattern` - a regular expression the input's contents must match in order to be valid
+- `size` - a number indicating how many characters wide the input field should be
+- `spellcheck` - controls whether or not to enable spell checking for the input field, or if the default spell checking configuration should be used
+.- `appearance` - enum
+  - outline (default)
+  - filled
+
+*Events*
+- `change: CustomEvent`
```
Comment (EisenbergEffect) [inline specs/search/search.md:56]: We should probably have both `change` and `input` events here.

**Review by EisenbergEffect — APPROVED** [review]

**Comment by chrisdholt** [comment]
> Seems that we already merged the component. Does this spec match what was implemented and can we merge it?

@eljefe223 can you make sure this is up to date and reflects where we currently are? Ideally we ensure these are merged before the component work is done :)

**Review by raghavthind2005 — APPROVED** [review]

**Pull request closed without merging**